x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/endpoint_details_tabs.tsx

```diff
@@ -27,8 +28,34 @@ interface EndpointDetailsTabs {
   id: string;
   name: string;
   content: JSX.Element;
+  route: string;
 }
 
+const EndpointDetailsTab = memo(
+  ({
+    tab,
+    isSelected,
+    handleTabClick,
+  }: {
+    tab: EndpointDetailsTabs;
+    isSelected: boolean;
+    handleTabClick: () => void;
+  }) => {
+    const setUrl = useNavigateByRouterEventHandler(tab.route);
+    const onClick = (e: React.MouseEvent<HTMLAnchorElement | HTMLButtonElement, MouseEvent>) => {
```

Comment:
Could you use useCallback here in order to memoize this function depending on setUrl and handleTabClick?

Reply:
Got it. I missed that somehow.

Reply:
done 07f9c3dfe689d8504beab02c8ecaab70ddbab41c